### PR TITLE
Camera & Gallery Integration

### PR DESCRIPTION
**Pull request by thisandagain**
Resolves GH-1564
Resolves GH-1563
Resolves GH-750

### CODE REVIEW

**Comment by thisandagain** [comment]
This is coming along pretty quickly. Camera activity and native compression is done. Just need to hook up the gallery and get the selector (JS) done:
screen shot 2015-05-13 at 11 14 14 pm

**Comment by thisandagain** [comment]
@ryanw-se Fixed!

**Comment by thisandagain** [comment]
Media gallery support (minus GIFs) is now complete as well. This just needs the interface from #1563 and this should be :white_check_mark: .

**Comment by thisandagain** [comment]
Ok. This is ready for review. @k88hudson R?

**Comment by k88hudson** [comment]
:clap: this is so awesome. will test it out first thing tmr

File: www_src/pages/element/element.jsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ render(React.createClass({
 
     if (typeof element === 'undefined') return (<div>Loading...</div>);
 
-    Editor = editors[params.editor] || editors[hash] || editors.text;
+    Editor = editors[params.editor] || editors[hash] || editors.image;
```
Comment (k88hudson) [inline www_src/pages/element/element.jsx]: any reason for this change?

**Comment by k88hudson** [comment]
Looks good to me, works well. You're right, there's definitely a slow down in the element editor -- but i'm almost 100% sure it's because `prefs`are getting updated on every state change.

File: www_src/pages/element/element.jsx (hunk 1 above)
Comment (thisandagain) [inline www_src/pages/element/element.jsx]: Missed from testing. Will revert.

File: app/src/main/java/mozilla/org/webmaker/activity/Element.java [hunk 2]
```diff
@@ -1,10 +1,78 @@
...
+     * @param resultCode
+     * @param data
+     */
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        Bitmap bitmap = null;
+
+        // Return if user exited without capturing / selecting an image
+        if (resultCode == 0) {
+            Log.v("DATAURI:RESULT", resultCode + "");
```
Comment (thisandagain) [inline app/src/main/java/mozilla/org/webmaker/activity/Element.java]: Don't need this log.

File: app/src/main/res/values/compression.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (thisandagain) [inline app/src/main/res/values/compression.xml]: Not used (for now)

File: app/src/main/java/mozilla/org/webmaker/activity/Element.java (hunk 2 above)
Comment (k88hudson) [inline app/src/main/java/mozilla/org/webmaker/activity/Element.java]: we should probably file a ticket to do a pass on logs, there are quite a few hanging around

File: www_src/static/img/take-photo.svg [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (thisandagain) [inline www_src/static/img/take-photo.svg]: Should be run through `svgo`

File: www_src/static/img/camera-gallery.svg [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (thisandagain) [inline www_src/static/img/camera-gallery.svg]: Should be run through `svgo`

File: app/src/main/java/mozilla/org/webmaker/activity/Element.java (hunk 2 above)
Comment (thisandagain) [inline app/src/main/java/mozilla/org/webmaker/activity/Element.java]: :+1: `logcat` is really noisy right now.

**Pull request merged**